esphome/components/seeed_mr24hpc1/seeed_mr24hpc1.h

```diff
@@ -0,0 +1,239 @@
+  SUB_BUTTON(reset)
+  SUB_BUTTON(custom_set_end)
+#endif
+#ifdef USE_SELECT
+  SUB_SELECT(scene_mode)
+  SUB_SELECT(unman_time)
+  SUB_SELECT(existence_boundary)
+  SUB_SELECT(motion_boundary)
+#endif
+#ifdef USE_NUMBER
+  SUB_NUMBER(sensitivity)
+  SUB_NUMBER(custom_mode)
+  SUB_NUMBER(existence_threshold)
+  SUB_NUMBER(motion_threshold)
+  SUB_NUMBER(motion_trigger)
+  SUB_NUMBER(motion_to_rest)
+  SUB_NUMBER(custom_unman_time)
+#endif
+
+ protected:
+  char c_product_mode_[PRODUCT_BUF_MAX_SIZE + 1];
+  char c_product_id_[PRODUCT_BUF_MAX_SIZE + 1];
+  char c_hardware_model_[PRODUCT_BUF_MAX_SIZE + 1];
+  char c_firmware_version_[PRODUCT_BUF_MAX_SIZE + 1];
+  uint8_t s_output_info_switch_flag_;
+  uint8_t sg_recv_data_state_;
+  uint8_t sg_frame_len_;
+  uint8_t sg_data_len_;
+  uint8_t sg_frame_buf_[FRAME_BUF_MAX_SIZE];
+  uint8_t sg_frame_prase_buf_[FRAME_BUF_MAX_SIZE];
+  int sg_start_query_data_;
+  bool check_dev_inf_sign_;
+  bool poll_time_base_func_check_;
+
+ public:
+  MR24HPC1Component() : PollingComponent(8000) {}  // The update() function is called every 8 seconds.
+  float get_setup_priority() const override { return esphome::setup_priority::LATE; }
+  void setup() override;
+  void update() override;
+  void dump_config() override;
+  void loop() override;
+  void r24_split_data_frame(uint8_t value);
+  void r24_parse_data_frame(uint8_t *data, uint8_t len);
+  void r24_frame_parse_open_underlying_information(uint8_t *data);
+  void r24_frame_parse_work_status(uint8_t *data);
+  void r24_frame_parse_product_information(uint8_t *data);
+  void r24_frame_parse_human_information(uint8_t *data);
+  void send_query(uint8_t *query, size_t string_length);
+
+  void get_heartbeat_packet();
+  void get_radar_output_information_switch();
+  void get_product_mode();
+  void get_product_id();
+  void get_hardware_model();
+  void get_firmware_version();
+  void get_human_status();
+  void get_human_motion_info();
+  void get_body_motion_params();
+  void get_keep_away();
+  void get_scene_mode();
+  void get_sensitivity();
+  void get_unmanned_time();
+  void get_custom_mode();
+  void get_existence_boundary();
+  void get_motion_boundary();
+  void get_spatial_static_value();
+  void get_spatial_motion_value();
+  void get_distance_of_static_object();
+  void get_distance_of_moving_object();
+  void get_target_movement_speed();
+  void get_existence_threshold();
+  void get_motion_threshold();
+  void get_motion_trigger_time();
+  void get_motion_to_rest_time();
+  void get_custom_unman_time();
+
+  void set_scene_mode(const std::string &state);
+  void set_underlying_open_function(bool enable);
+  void set_sensitivity(uint8_t value);
+  void set_reset();
+  void set_unman_time(const std::string &time);
+  void set_custom_mode(uint8_t mode);
+  void set_custom_end_mode();
+  void set_existence_boundary(const std::string &value);
+  void set_motion_boundary(const std::string &value);
+  void set_existence_threshold(int value);
```

Comment:
Please use the correct limiting type for arguments and variables to lessen the amount of ram used by this component.
```suggestion
  void set_existence_threshold(uint8_t value);
```